### ISSUE TITLE
ssd 在 coco 上训练，出现 loss 为 Nan，test map 过低的问题

### ISSUE THREAD

**Issue by Aneureka** [PaddlePaddle/models#3181]
训练的命令是
```shell
python3 -u train.py --dataset coco2017 --data_dir datasets/coco --batch_size 32 --pretrain
ed_model pretrained/ssd_mobilenet_v1_coco --learning_rate 0.001
```
加载的是官方的 mobileNet + ssd 与训练模型，
然后在 epoch == 4 的时候就出现 loss 为 Nan 的问题了
![img](http://ww4.sinaimg.cn/large/006y8mN6gy1g6b4pzjzr4j30t60p0ad2.jpg)

训练的时候 best_model 对应的 map 有 0.3 左右，但测试的时候只有 0.02
已经考虑到 91 -> 81 的问题...

**Comment by yghstill** [comment]
@Aneureka 
（1）是几卡训练的？可尝试下4卡，每卡batch size=32，lr=0.001训练。如果卡数减半需要降低lr或者batch size翻倍。
（2）如果还是出现nan情况，请参考 #3130 最新回复。
（3）也欢迎使用[PaddleDetection](https://github.com/PaddlePaddle/models/tree/develop/PaddleCV/PaddleDetection)来训练，具体配置文件可参考[config文件](https://github.com/PaddlePaddle/models/pull/3188)

**Comment by Aneureka** [comment]
> @Aneureka
> （1）是几卡训练的？可尝试下4卡，每卡batch size=32，lr=0.001训练。如果卡数减半需要降低lr或者batch size翻倍。
> （2）如果还是出现nan情况，请参考 #3130 最新回复。
> （3）也欢迎使用[PaddleDetection](https://github.com/PaddlePaddle/models/tree/develop/PaddleCV/PaddleDetection)来训练，具体配置文件可参考[config文件](https://github.com/PaddlePaddle/models/pull/3188)

谢谢回答~ 之前发生这个情况是在自己的机子上单卡训练的，lr=0.001，batch_size=32，现在用 aistudio 训练就不会 *太早* 发生这种情况了，但在 loss 降到 4 左右会变为 None，目前的解决方案是 loss 变成 None 之后直接降低 lr 。。

**Comment by yghstill** [comment]
@Aneureka 请参考[#3130下回复](https://github.com/PaddlePaddle/models/issues/3130#issuecomment-524693930)，修改代码后再训。
另外MobileNet-SSD在COCO上复现可使用在VOC数据集上训练得到的权重作为预训练模型，去除multi_box_head部分的权重。经测试在COCO上box ap可以达到22.5，供参考。使用imagenet-pretrain的MobileNet-SSD模型正在验证，检测库中（PaddleDetection）待发布

**Comment by PJSSABER** [comment]
请问一下 91 -> 81 这个问题应该怎么解决啊

**Comment by yghstill** [comment]
> 请问一下 91 -> 81 这个问题应该怎么解决啊
@PJSSABER 请参考[PaddleDetection](https://github.com/PaddlePaddle/models/blob/develop/PaddleCV/PaddleDetection/ppdet/utils/coco_eval.py#L380)
